compiler/fir/raw-fir/raw-fir.common/src/org/jetbrains/kotlin/fir/builder/BaseFirBuilder.kt

```diff
@@ -214,13 +214,25 @@ abstract class BaseFirBuilder<T>(val baseSession: FirSession, val context: Conte
         }
     }
 
-    fun FirLoopBuilder.configure(generateBlock: () -> FirBlock): FirLoop {
+    fun FirLoopBuilder.prepareTarget() {
         label = context.firLabels.pop()
         val target = FirLoopTarget(label?.name)
         context.firLoopTargets += target
+    }
+
+    fun FirLoopBuilder.generateCondition(generateCondition: () -> FirExpression): FirLoopBuilder {
+        condition = generateCondition()
+        return this
+    }
+
+    fun FirLoopBuilder.generateBlock(generateBlock: () -> FirBlock): FirLoopBuilder {
         block = generateBlock()
+        return this
+    }
+
+    fun FirLoopBuilder.buildAndBind(): FirLoop {
```

Comment:
Looks like `generateBlock()` and `buildAndBind()` are always called one after another, maybe you can just combine them.

Reply:
Right. Merged and named it back to `configure` so that the changes are minimal.